src/cargo/sources/registry/mod.rs

```diff
@@ -752,7 +752,13 @@ impl<'gctx> Source for RegistrySource<'gctx> {
         if let Some((_, requested)) = self
             .source_id
             .precise_registry_version(dep.package_name().as_str())
-            .filter(|(c, _)| req.matches(c))
+            .filter(|(c, _)| {
+                if self.gctx.cli_unstable().unstable_options {
```

Comment:
I don't know how `match_prerelease` will evolve, so it's better to have a check here maybe?

```suggestion
            .filter(|(c, to)| {
                if to.is_prerelease() && self.gctx.cli_unstable().unstable_options {
```

Reply:
Your concern makes sense.